### ISSUE TITLE
TypeError: Cannot serialize non-str key None

### ISSUE THREAD

**Issue by TobyStoe** [humbertogontijo/homeassistant-roborock#116]
After updating HA-Core

Current Versions:
Home Assistant 2023.1.0
Supervisor 2022.12.1
Operating System 9.4
Frontend 20230104.0

`Error setting up entry Roborock S7 MaxV for roborock
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 382, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/roborock/__init__.py", line 70, in async_setup_entry
    home_data = HomeData(await api_client.get_home_data(user_data))
  File "/config/custom_components/roborock/api/api.py", line 420, in get_home_data
    home_id_response = await home_id_request.request(
  File "/config/custom_components/roborock/api/api.py", line 64, in request
    async with session.request(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1138, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 557, in _request
    resp = await req.send(conn)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 669, in send
    await writer.write_headers(status_line, self.headers)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/http_writer.py", line 130, in write_headers
    buf = _serialize_headers(status_line, headers)
  File "aiohttp/_http_writer.pyx", line 132, in aiohttp._http_writer._serialize_headers
  File "aiohttp/_http_writer.pyx", line 109, in aiohttp._http_writer.to_str
TypeError: Cannot serialize non-str key None`

**Comment by eugenet8k** [comment]
Yeah, version 0.21 gives me these:

```
Logger: homeassistant.components.vacuum
Source: custom_components/roborock/device.py:65 
Integration: Vacuum ([documentation](https://www.home-assistant.io/integrations/vacuum), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+vacuum%22)) 
First occurred: 11:10:54 AM (4 occurrences) 
Last logged: 11:11:38 AM

Error adding entities for domain vacuum with platform roborock
Error while setting up roborock platform for vacuum
Traceback (most recent call last):
  File "/Users/pi/.homeassistant-venv/lib/python3.9/site-packages/homeassistant/helpers/entity_platform.py", line 428, in async_add_entities
    await asyncio.gather(*tasks)
  File "/Users/pi/.homeassistant-venv/lib/python3.9/site-packages/homeassistant/helpers/entity_platform.py", line 692, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/Users/pi/.homeassistant-venv/lib/python3.9/site-packages/homeassistant/helpers/entity.py", line 789, in add_to_platform_finish
    self.async_write_ha_state()
  File "/Users/pi/.homeassistant-venv/lib/python3.9/site-packages/homeassistant/helpers/entity.py", line 545, in async_write_ha_state
    self._async_write_ha_state()
  File "/Users/pi/.homeassistant-venv/lib/python3.9/site-packages/homeassistant/helpers/entity.py", line 585, in _async_write_ha_state
    attr.update(self.state_attributes or {})
  File "/Users/pi/.homeassistant/custom_components/roborock/vacuum.py", line 302, in state_attributes
    data[ATTR_ERROR] = self.error
  File "/Users/pi/.homeassistant/custom_components/roborock/vacuum.py", line 358, in error
    return self.translate(self.translation_key, ATTR_ERROR, error_code)
  File "/Users/pi/.homeassistant/custom_components/roborock/device.py", line 65, in translate
    return self.coordinator.translation.get(translation_key).get(attr).get(str(value))
AttributeError: 'NoneType' object has no attribute 'get'
```

```
Logger: homeassistant.components.sensor
Source: custom_components/roborock/device.py:65 
Integration: Sensor ([documentation](https://www.home-assistant.io/integrations/sensor), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+sensor%22)) 
First occurred: 11:10:54 AM (2 occurrences) 
Last logged: 11:11:38 AM

Error while setting up roborock platform for sensor
Traceback (most recent call last):
  File "/Users/pi/.homeassistant-venv/lib/python3.9/site-packages/homeassistant/helpers/entity_platform.py", line 281, in _async_setup_platform
    await asyncio.shield(task)
  File "/Users/pi/.homeassistant/custom_components/roborock/sensor.py", line 250, in async_setup_entry
    RoborockSensor(
  File "/Users/pi/.homeassistant/custom_components/roborock/sensor.py", line 272, in __init__
    self._attr_native_value = self._determine_native_value()
  File "/Users/pi/.homeassistant/custom_components/roborock/sensor.py", line 331, in _determine_native_value
    native_value = self.translate(
  File "/Users/pi/.homeassistant/custom_components/roborock/device.py", line 65, in translate
    return self.coordinator.translation.get(translation_key).get(attr).get(str(value))
AttributeError: 'NoneType' object has no attribute 'get'
```


```
Logger: homeassistant.components.camera
Source: custom_components/roborock/camera.py:56 
Integration: Camera ([documentation](https://www.home-assistant.io/integrations/camera), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+camera%22)) 
First occurred: 11:10:54 AM (1 occurrences) 
Last logged: 11:10:54 AM

Error while setting up roborock platform for camera
Traceback (most recent call last):
  File "/Users/pi/.homeassistant-venv/lib/python3.9/site-packages/homeassistant/helpers/entity_platform.py", line 281, in _async_setup_platform
    await asyncio.shield(task)
  File "/Users/pi/.homeassistant/custom_components/roborock/camera.py", line 56, in async_setup_entry
    image_config = config_entry.options[CONF_MAP_TRANSFORM]
KeyError: 'map_transformation'
```

**Comment by nathanfiscus** [comment]
Have you tried clicking the "Configure" button, submitting the configuration, and then reloading the integration?  I think the configuration is missing a default or a fallback for the map_transformation options.  I was getting the same error and once I reconfigured the integration I didn't have any more errors in my logs.

**Comment by Caine72** [comment]
> Have you tried clicking the "Configure" button, submitting the configuration, and then reloading the integration? I think the configuration is missing a default or a fallback for the map_transformation options. I was getting the same error and once I reconfigured the integration I didn't have any more errors in my logs.

@nathanfiscus  Had this specific issue. 
Your proposed solution fixed it for me!

My steps.
Changed the map rotation value. 
Saved.
As soon it was submitted, the config the camera entity was created.
Changed the value back.. as I did not want that map rotation.
Entity still there.
  
So indeed seems like it's missing a default value.

**Comment by TobyStoe** [comment]
I tried to change every value of the config, but the error still remains after reloading.
Same Error-Message

**Comment by TobyStoe** [comment]
Workaround worked for me:

Delete the Integration completly and re-installing it again. After config it worked again

**Comment by SVNKoch** [comment]
for me reinstalling was the only way to get it fixed too (updated from I think 0.0.17)
